### DISCUSSION TITLE
Modules system

### DISCUSSION THREAD

**Discussion by Chewbaka69** [lanedirt/OGameX#113] [Ideas]
Maybe a modules system can be nice.
To start i think we need like hook system and stuff like but a simple system like:
Listing all the modules and states in a config var to allow dev to modify pages and controller

Can be really a nice feature :)

**Comment by lanedirt** [comment]
A modules system sounds interesting, it could surely help to customize servers. It didn't cross my mind before.

Do you have any ideas or examples on what kind of modules you would expect the game to have?

**Comment by Chewbaka69** [comment]
Yes sure:
- Lottery system
- Tournament
- Daily Achivement
- Daily Connexion (daily bonus)
- Alliance Bank
- Alliance Planete
- Planet type (differente type of planete can increase, metaln, deterium, energy etc., you can change one type freely by planete in addition of the production by the position in the galaxy)
- Market place (every player can sell robots or ressources)
- Bounty hunter (you can pay someone to attack an other player, with a percentage of destruction mandatory to win the contract)
- Banned list
- Empire view
- Fleet trader (to sell your fleet)
- Asteroid (some asteroid spawn every X times in the galaxy and you need to fight the fleet on it to capture it, and each type of asteroid produce different type of ressource)

You can find more example maybe here: 
https://2moons.de/index.php?board-list/
or on my screen

![image](https://github.com/lanedirt/OGameX/assets/3535011/25d15c9d-8790-49a6-9e52-9af1b23af820)
